docs/recipes/developing-and-testing/debug-safe-app.md

```diff
@@ -85,23 +85,54 @@ Click the "Add Configuration..." button and choose **Launch .NET Core Console Ap
 The only change required is to point it at the Server application, by replacing the `program` line with this:
 
 ```json
-"program": "${workspaceFolder}/src/Server/bin/Debug/netcoreapp3.1/Server.dll",
+"program": "${workspaceFolder}/src/Server/bin/Debug/net5.0/Server.dll",
 ```
 
 #### 3. Configure a build task
 * From the Command Palette, choose `Configure Task`.
 * Select `Create tasks.json file from template`. This will show you a list of pre-configured templates.
 * Select `.NET Core`.
+* Update the build directory using `"options": {"cwd": "src/Server},` as shown below:
+
+```json
+{
+    // See https://go.microsoft.com/fwlink/?LinkId=733558
+    // for the documentation about the tasks.json format
+    "version": "2.0.0",
+    "tasks": [
+        {
+            "label": "build",
+            "command": "dotnet",
+            "type": "shell",
+            "options": {"cwd": "src/Server"}, 
+            "args": [
+                "build",
+                "debug-pt3.sln",
+                // Ask dotnet build to generate full paths for file names.
+                "/property:GenerateFullPaths=true",
+                // Do not generate summary otherwise it leads to duplicate errors in Problems panel
+                "/consoleloggerparameters:NoSummary"
+            ],
+            "group": "build",
+            "presentation": {
+                "reveal": "silent"
+            },
+            "problemMatcher": "$msCompile"
+        }
+    ]
+}
+```
+
 
 #### 4. Debug the Server
 Either hit F5 or open the Debugging pane and press the Play button to build and launch the Server with the debugger attached.
 Observe that the Debug Console panel will show output from the server. The server is now running and you can set breakpoints and view the callstack etc.
 
 #### 5. Debug the Client
-* Start the Client using e.g. `npm run start`.
+* Start the Client using `dotnet fable watch src/Client --run webpack-dev-server`.
 * Open the Command Palette and run `Debug: Open Link`.
 * When prompted for a url, type `http://localhost:8080/`. This will launch a browser which is pointed at the URL and connect the debugger to it.
-* You can now set breakpoints in your Client within VS Code.
+* You can now set breakpoints in the generated `.fs.js` files within VS Code.
```

Comment:
We just tried firing up a new Safe V3 app, going through the launch settings setup as described in this recipe, and adding the `-s` when we launch fable, but we still only get breakpoints in the .fs.js files. 

As you say, we did have breakpoints in F# previously - is there anything else we should try?

Reply:
The "sourceMaps" option may be needed in the launch.json too. With a config like the one in [the comment below](https://github.com/SAFE-Stack/docs/pull/223#discussion_r654116176) I managed to set breakpoints in F# code. This was in my project, I will try with a Safe V3 app.

Reply:
Ok, my bad. I forgot we need to add source-map-loader to the webpack.config.js. However, when I did that to a SAFE v3 app I got conflicts with the versions, I tried updating Webpack and got more errors 😢 so we'll have to leave this until I (or someone else) has time to try to update SAFE to Webpack 5 (I got it working in my project, not sure why it failed with SAFE).

Reply:
Perhaps this is something we can add as a post-go-live? Unless it's something that'll be quick to do.

I really want to get V3 out this week - it's been dragging on and there's a lot of good stuff that people can start benefitting from immediately.

Reply:
Yes, let's do that later. I agree we should have V3 out as soon as possible. I also updated the Fable docs recently pointing to the SAFE template.

Reply:
I just tried it and it actually works fine.
This is using a standard unmodified `v3.0.0-beta004`, just following the vscode part of the recipe but adding the `-s` when launching fable.

I can add and hit breakpoints to client/shared fs files. Did not even have to add a config for the client in launch.json nor change anything with webpack.

There were a couple of sticking points while following the recipe:
 - `Debug: Open launch.json` isn't an option in the command pallete. I couldn't find any vscode commands that do that, so I just clicked *create a launch.json file* in the debug tab instead.
 - When I select `.NET Core`, it shows an error "Cannot create .NET debug configurations. The OmniSharp server is still initializing or has exited unexpectedly". This didn't seem to matter and I just ignored it.
 - In the samples tasks.json build task shown, it includes `debug-pt3.sln` in the args. When I created the configuration, it didn't include the sln as an arg, and if you add it in conjunction with the cwd option as shown, it will actually break the build because the sln file is not in src/Server. Should probably just remove that line `debug-pt3.sln` from the sample shown.
 - As I had jumped straight into the recipe without running anything else, when I started the client it failed to start webpack-dev-server. This was fixed by running `dotnet run -- installclient`. Perhaps running `npm install` could be added as a prerequisite

Now, when I added the `-s` to the fable command, it didn't actually create the .fs.js.map files. I assume this is because of some cache from when it was run without the `-s`. Saving the fs file triggered it to output the .fs.js.map file. Is this a bug?

Also we should probably add `*.fs.js.map` to the `.gitignore`.